concordium-contracts-common-derive/src/derive.rs

```diff
@@ -1148,3 +1459,74 @@ fn schema_type_fields(fields: &syn::Fields) -> syn::Result<proc_macro2::TokenStr
         syn::Fields::Unit => Ok(quote! { concordium_std::schema::Fields::None }),
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    /// Test parsing for bound attribute when using syntax for sharing.
+    #[test]
+    fn test_parse_shared_bounds() {
+        let ast: syn::ItemStruct = syn::parse_quote! {
+            #[concordium(bound = "T: B")]
+            struct MyStruct{}
+        };
+
+        let parsed = ContainerAttributes::try_from(ast.attrs.as_slice())
+            .expect("Failed to parse container attributes");
+
+        assert!(parsed.deserial_bounds().is_some(), "Failed to add shared bound");
+        assert!(parsed.serial_bounds().is_some(), "Failed to add shared bound");
+        assert!(parsed.schema_type_bounds().is_some(), "Failed to add shared bound");
+    }
+
+    /// Test parsing for bound attribute when using syntax for Deserial
+    /// explicit.
+    #[test]
+    fn test_parse_deserial_explicit_bounds() {
+        let ast: syn::ItemStruct = syn::parse_quote! {
+            #[concordium(bound(deserial  = "T: B"))]
+            struct MyStruct{}
+        };
+
+        let parsed = ContainerAttributes::try_from(ast.attrs.as_slice())
+            .expect("Failed to parse container attributes");
+
+        assert!(parsed.deserial_bounds().is_some(), "Failed to add explicit bound");
+        assert!(parsed.serial_bounds().is_none(), "Unexpected bound added for Serial");
+        assert!(parsed.schema_type_bounds().is_none(), "Unexpected bound added for SchemaType");
+    }
+
+    /// Test parsing for bound attribute when using syntax for Serial explicit.
+    #[test]
+    fn test_parse_serial_explicit_bounds() {
+        let ast: syn::ItemStruct = syn::parse_quote! {
+            #[concordium(bound(serial  = "T: B"))]
+            struct MyStruct{}
+        };
```

Comment:
This seems a bit of an odd test. I would not expect this to compile (I know this is not being tested here).

But it is an odd test case that the struct does not have a generic.

Reply:
I'll make it more like a real struct then